docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+Hypothesis CSS Style Guide
+==========================
+
+Hypothesis' CSS generally follows the principles of [Trello's CSS Style Guide](https://github.com/trello/trellisheets)
+for writing maintainable styling. In particular:
+
+ * Use simple, non-nested class selectors where possible
+ * Organize CSS into components. Unlike Trello, we use BEM-style naming (see below) and do not indent selectors
```

Comment:
What's the motivation for using BEM-style naming instead of just using Trello's style?

Quite a lot of the Trello CSS guide is describing their naming style / the whole thing is sort of centered around the naming style, I think it makes it a little difficult to follow the Trello style but use a different naming style.

Might be worth it if there's a significant advantage?